vNext/channels/applicationinsights-channel-js/src/EnvelopeCreator.ts

```diff
@@ -57,6 +57,9 @@ export abstract class EnvelopeCreator {
     protected static createEnvelope<T>(logger: IDiagnosticLogger, envelopeType: string, telemetryItem: ITelemetryItem, data: Data<T>): IEnvelope {
         let envelope = new Envelope(logger, data, envelopeType);
         envelope.iKey = telemetryItem.iKey;
+        if (data.baseData && (data.baseData as any).time) {
```

Comment:
This time property only applies to dependency events but you will check for it for every single other event, is there a better place to add this code?

Reply:
Move this to after line 239. Once u get the envelope back for dependency, update timestamp with baseData.time (or baseData.requestSentTime as you name it)

EnvelopeCreator.createEnvelope<RemoteDependencyData>(logger, RemoteDependencyData.envelopeType, telemetryItem, data);

Reply:
moved to 239